src/context-parser-handlebars.js

```diff
@@ -404,17 +431,83 @@ ContextParserHandlebars.prototype.analyzeAst = function(ast, contextParser, char
+                    exceptionObj = new ContextParserHandlebarsException(msg, this._fileName, this._lineNo, this._charNo);
+                    handlebarsUtils.handleError(exceptionObj, true);
+                }
+
+                namespace = this._config._namespace === ''? '': this._config._namespace + '/';
+                key = namespace + partialName;
+
+                var keyReg = new RegExp("^" + namespace + "SJST\d+_");
+                // if the key is our generated partial template, there is no need to reprocess it again,
+	        // just put back the partial expression is fine
+                if (keyReg.test(key)) {
+                    output += node.content;
+                } else {
+                    partialContent = this._config._rawPartialsCache[key];
+
+                    if (partialContent === undefined) {
+                        output += node.content; // this._config._strictMode=true will throw
+
+                        if (!this._config._strictMode || !this._config._disablePartialParsing) {
+                            msg = (this._config._strictMode? '[ERROR]' : '[WARNING]') + " SecureHandlebars: Fail to load the partial " + node.content + ' content!';
+                            msg += "\nPlease follow this URL to resolve - https://github.com/yahoo/secure-handlebars#warnings-and-workarounds";
+                            exceptionObj = new ContextParserHandlebarsException(msg, this._fileName, this._lineNo, this._charNo);
+                            handlebarsUtils.handleError(exceptionObj, this._config._strictMode);
+                        }
+
+                    } else {
+                        // get the html state number right the parital is called, and analyzed
+                        enterState = parser.getCurrentState();
+
+                        // while analyzing the partial, use the current parser (possibly forked) and disable printChar
+                        partialContent = this.analyzeContext(partialContent, {
+                            contextParser: parser,
+                            disablePrintChar: true
+                        });
+
+                        if (this._config._enablePartialCombine) {
+                            output += partialContent;
+                        } else {
+                            // https://github.com/yahoo/secure-handlebars/blob/master/src/handlebars-utils.js#L76
+                            var pattern = /^(\{\{~?>\s*)([^\s!"#%&'\(\)\*\+,\.\/;<=>@\[\\\]\^`\{\|\}\~]+)(.*)/,
+                                newPartialName = 'SJST'+enterState+'_'+partialName;
+                    
+                            // rewrite the partial name, that is suffixed with the in-state
+                            output += node.content.replace(pattern, function(m, p1, p2, p3) {
+                                return p1 + newPartialName + p3;
+                            });
+
+                            key = namespace + newPartialName;
```

Comment:
should newPartialName contains namespace too? otherwise, how can the cache be leveraged?